dm/worker/worker.go

```diff
@@ -198,6 +199,16 @@ func (w *Worker) EnableRelay() (err error) {
 		w.l.Warn("already enabled relay")
 		return nil
 	}
+	// we need update worker source config from etcd first
+	// because the configuration of the relay part of the data source may be changed via scheduler.UpdateSourceCfg
+	sourceCfg, _, err := ha.GetRelayConfig(w.etcdClient, w.name)
+	if err != nil {
+		return err
+	}
+	failpoint.Inject("MockGetSourceCFGFromETCD", func(_ failpoint.Value) {
+		sourceCfg = w.cfg
+	})
+	w.cfg = sourceCfg
```

Comment:
can we make sure the rest part of source config is not changed?

Reply:
add this check in `UpdateSourceCfg .checkSourceCfgCanUpdated `  https://github.com/pingcap/dm/pull/2083/commits/f66260e4eb31b9b50108a2347d1f49ae54256d91